src/main/java/com/coreos/jetcd/resolver/AbstractEtcdNameResolver.java

```diff
@@ -0,0 +1,59 @@
+package com.coreos.jetcd.resolver;
+
+import java.net.URI;
+import java.util.Collections;
+import java.util.List;
+import javax.annotation.concurrent.GuardedBy;
+
+import io.grpc.Attributes;
+import io.grpc.NameResolver;
+import io.grpc.ResolvedServerInfo;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+import static com.google.common.base.Preconditions.checkState;
+
+abstract class AbstractEtcdNameResolver extends NameResolver {
```

Comment:
I believe we should reuse this `AbstractEtcdNameResolver` with `SimpleEtcdNameResolver`.